### PR TITLE
Refactored url and content type handling

### PR DESCRIPTION
**Pull request by danrot**
This fixes the disappearance of entire route trees. Also the handling of content types have been improved, so that it will be easier to find errors in production systems.

__tasks:__

- [ ] test coverage

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| Related PRs      | none
| BC breaks        | hopefully none, every introduced BC break would be a bug
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Component/Content/Document/Subscriber/StructureSubscriber.php [hunk 1]
```diff
@@ -258,17 +258,13 @@ private function mapContentToNode($document, NodeInterface $node, $locale)
...
-                );
-            } catch (\Exception $e) {
-            }
+            $contentType->remove(
+                $node,
+                $legacyProperty,
+                $webspaceName,
+                $locale,
+                null
+            );
```
Comment (danrot) [inline src/Sulu/Component/Content/Document/Subscriber/StructureSubscriber.php]: Tried to remove a property by hand in the phpcr shell, and saved a page again, was still working. So I think we really do not need to check any exceptions here. /cc @wachterjohannes @chirimoya

File: src/Sulu/Component/Content/Types/Rlp/Mapper/PhpcrMapper.php [hunk 2]
```diff
@@ -449,7 +455,6 @@ function ($resourceLocator, NodeInterface $historyNode) use (
                 ) {
                     // delete history nodes
                     $this->deleteByNode($historyNode, $session, $webspaceKey, $languageCode, $segmentKey);
-                    $session->save();
```
Comment (danrot) [inline src/Sulu/Component/Content/Types/Rlp/Mapper/PhpcrMapper.php]: Any suggestions where this might have been necessary? Everything I tested was working so far. /cc @wachterjohannes @chirimoya

File: src/Sulu/Component/Content/Types/ResourceLocator.php [hunk 3]
```diff
@@ -186,7 +185,6 @@ public function remove(
         $languageCode,
         $segmentKey = null
     ) {
-        $this->strategy->deleteByPath($property->getValue(), $webspaceKey, $languageCode, $segmentKey);
```
Comment (danrot) [inline src/Sulu/Component/Content/Types/ResourceLocator.php]: This line caused the deletion of an entire subtree.

**Comment by danrot** [comment]
@wachterjohannes @chirimoya I think it would be a good idea if you both test this, since it is really a critical change. Would that be possible?

File: tests/Sulu/Component/Content/Rlp/Mapper/PhpcrMapperTest.php [hunk 4]
```diff
@@ -450,6 +450,24 @@ public function testDeleteByPath()
         $this->assertFalse($rootNode->hasNode('test/news-1/sub-2'));
     }
 
+    /**
+     * The deleteByPath method would delete the entire route tree for the given language, if an empty path is passed.
+     */
+    public function testDeleteByPathWithEmptyPath()
```
Comment (wachterjohannes) [inline tests/Sulu/Component/Content/Rlp/Mapper/PhpcrMapperTest.php]: maybe also add a test for null? i would also prefer to combine this three tests to one with a dataProvider!

**Comment by wachterjohannes** [comment]
@danrot i have not found any problems with this change! please fix the comment before then i will merge it!

File: tests/Sulu/Component/Content/Rlp/Mapper/PhpcrMapperTest.php (hunk 4 above)
Comment (danrot) [inline tests/Sulu/Component/Content/Rlp/Mapper/PhpcrMapperTest.php]: Done

**Comment by wachterjohannes** [comment]
ok i will wait for tests and then merge it!
